### PR TITLE
Fleet Aggregate Player Tracking Logic

### PR DESCRIPTION
**Pull request by markmandel**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespace from that line:
>
> /kind breaking
> /kind bug
> /kind cleanup
> /kind documentation

/kind feature

> /kind hotfix

**What this PR does / Why we need it**:

Implementation of aggregation of Player Tracking values at the Fleet and GameServerSet levels.

**Which issue(s) this PR fixes**:

Work on #1033

**Special notes for your reviewer**:

Includes both unit and e2e tests.

This completes the logic for player tracking :raised_hands: 

Docs are the final piece left :muscle:

### CODE REVIEW

File: pkg/gameserversets/controller.go [hunk 1]
```diff
@@ -606,5 +606,21 @@ func computeStatus(list []*agonesv1.GameServer) agonesv1.GameServerSetStatus {
 		}
 	}
 
+	if runtime.FeatureEnabled(runtime.FeaturePlayerTracking) {
+		// to make this code simpler, we will loop around the gs list twice.
+		status.Players = &agonesv1.AggregatedPlayerStatus{}
+		for _, gs := range list {
+			if gs.ObjectMeta.DeletionTimestamp.IsZero() &&
+				(gs.Status.State == agonesv1.GameServerStateReady ||
```
Comment (pooneh-m) [inline pkg/gameserversets/controller.go:614]: If a GS is ready or reserved, does it still have any player?

Comment (markmandel) [reply pkg/gameserversets/controller.go:614]: Good question :thinking: 

In theory it _might_, say leftover from a previous session, and the GameServer has gone back into a pool, and the players are still connected.

I figured it made sense to err on the side of more flexible.

File: pkg/fleets/controller.go [hunk 2]
```diff
@@ -531,6 +531,17 @@ func (c *Controller) updateFleetStatus(fleet *agonesv1.Fleet) error {
 		fCopy.Status.ReservedReplicas += gsSet.Status.ReservedReplicas
 		fCopy.Status.AllocatedReplicas += gsSet.Status.AllocatedReplicas
 	}
+	if runtime.FeatureEnabled(runtime.FeaturePlayerTracking) {
+		// to make this code simpler, we will loop around the gsSet list twice.
```
Comment (roberthbailey) [inline pkg/fleets/controller.go:535]: simpler while the feature gate is in place? Do you want to add a TODO to consolidate this with the above loop when the feature graduates to ga?

Comment (markmandel) [reply pkg/fleets/controller.go:535]: > simpler while the feature gate is in place? 

Yeah exactly. Let me be explicit about this. Excellent idea on the TODO, doing that now.

**Pull request merged**